test/Frontend/skip-function-bodies.swift

```diff
@@ -0,0 +1,378 @@
+// RUN: %empty-directory(%t)
+
+// Check -emit-ir and -c are invalid when skipping function bodies
+// RUN: not %target-swift-frontend -emit-ir %s -experimental-skip-non-inlinable-function-bodies %s 2>&1 | %FileCheck %s --check-prefix ERROR
+// RUN: not %target-swift-frontend -c %s -experimental-skip-non-inlinable-function-bodies %s 2>&1 | %FileCheck %s --check-prefix ERROR
+// RUN: not %target-swift-frontend -emit-ir %s -experimental-skip-all-function-bodies %s 2>&1 | %FileCheck %s --check-prefix ERROR
+// RUN: not %target-swift-frontend -c %s -experimental-skip-all-function-bodies %s 2>&1 | %FileCheck %s --check-prefix ERROR
+// ERROR: -experimental-skip-*-function-bodies do not support emitting IR
+
+// Warn when trying to build SwiftONoneSupport with skip non-inlinable
+// function bodies enabled
+// RUN: %target-swift-frontend -typecheck -experimental-skip-non-inlinable-function-bodies -module-name SwiftOnoneSupport %s 2>&1 | %FileCheck %s --check-prefix WARNING
+// WARNING: module 'SwiftOnoneSupport' cannot be built with -experimental-skip-non-inlinable-function-bodies; this option has been automatically disabled
+
+// Check skipped bodies are neither typechecked nor SILgen'd
+// RUN: %target-swift-frontend -emit-sil -emit-sorted-sil -experimental-skip-non-inlinable-function-bodies -debug-forbid-typecheck-prefix INLINENOTYPECHECK %s -o %t/Skip.noninlinable.sil
+// RUN: %target-swift-frontend -emit-sil -emit-sorted-sil -O -experimental-skip-all-function-bodies -debug-forbid-typecheck-prefix ALLNOTYPECHECK %s -o %t/Skip.all.sil
```

Comment:
`-debug-forbid-typecheck-prefix` is a nice addition to that test, I didn't know we had this flag.

Reply:
Heh, the "I didn't know" is me almost every day at the moment :D!